src/operationDefinitions/types.ts

```diff
@@ -5,20 +5,36 @@
 
 import { Router } from 'express';
 import { SystemOperation, TypeOperation } from 'fhir-works-on-aws-interface';
+import ResourceHandler from '../router/handlers/resourceHandler';
 
 export interface OperationDefinitionImplementation {
     /**
      * url of the corresponding OperationDefinition resource
      */
-    canonicalUrl: string;
+    readonly canonicalUrl: string;
+
+    /**
+     * url path used to invoke the operation
+     */
+    readonly path: string;
+
+    /**
+     * methods supported by this operation e.g GET, POST
+     */
+    readonly httpMethods: string[];
+
+    /**
+     * FHIR resourceType that is affected by this operation
+     */
+    readonly targetResourceType: string;
```

Comment:
Are there situations where it is a global operation? could we add that layer in now or perhaps wait until that situation arises?